Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Quote.hs

```diff
@@ -112,11 +114,33 @@ findAllByRequestId searchRequestId = Esq.buildDType $ do
     pure (quote, mbTripTerms, mbRentalSlab, mbDriverOffer, mbspecialZoneQuote)
   catMaybes <$> mapM buildFullQuote fullQuoteTs
 
-findAllByEstimateId :: Transactionable m => Id Estimate -> m [Quote]
-findAllByEstimateId estimateId = buildDType $ do
-  fullQuoteTs <- Esq.findAll' $ do
-    (quote :& mbTripTerms :& mbRentalSlab :& mbDriverOffer :& mbspecialZoneQuote) <- from fullQuoteTable
-    where_ $
-      mbDriverOffer ?. DriverOfferEstimateId ==. just (val $ toKey estimateId)
-    pure (quote, mbTripTerms, mbRentalSlab, mbDriverOffer, mbspecialZoneQuote)
-  catMaybes <$> mapM buildFullQuote fullQuoteTs
```

Comment:
```
findAllDriverOfferQuotesByEstimateId :: Transactionable m => Id Estimate -> m [Quote]
findAllDriverOfferQuotesByEstimateId estimateId = buildDType $ do
  fullQuoteTs <- Esq.findAll' $ do
    (quote :& driverOffer) <-
      from $
        table @QuoteT
          `innerJoin` table @DriverOfferT
            `Esq.on` ( \(quote :& driverOffer) ->
                         quote ^. QuoteDriverOfferId ==. Esq.just (driverOffer ^. DriverOfferTId)
                     )
    where_ $
      quote ^. QuoteFareProductType ==. val DFP.DRIVER_OFFER
        &&. driverOffer ^. DriverOfferEstimateId ==. val (toKey estimateId)
    pure (quote, driverOffer)
  catMaybes <$> mapM buildFullQuote ((\(quoteT, driverOfferT) -> (quoteT, Nothing, Nothing, Just driverOfferT, Nothing)) <$> fullQuoteTs)
```
 This should work better. 
We really should deprecate this leftJoin usage in "either this or that" cases like this, every query in this file doing too much load on DB. This particular select is obviously DriverOffer one, so we can use join here, but in other generic selects from this file it might work faster as separate selects and we should refactor it. @NikithShetty @hkmangla

Reply:
@DonDokidon we are moving towards removing joins altogether https://github.com/nammayatri/nammayatri/pull/460#issuecomment-1500510233

Reply:
Merging this PR for now